Source/Core/Core/AchievementManager.cpp

```diff
@@ -262,6 +266,129 @@ bool AchievementManager::IsHardcoreModeActive() const
   return rc_client_is_processing_required(m_client);
 }
 
+bool AchievementManager::VerifyPatchHash(const Common::SHA1::Digest& digest) const
+{
+  std::string sys_path = File::GetSysDirectory() + DIR_SEP;
+
+  File::IOFile list_file;
+  list_file.Open(fmt::format("{}{}", sys_path, APPROVED_LIST_FILENAME), "rb");
```

Comment:
You need to check whether the File.Open actually succeded.

Reply:
Also there's no need for the `sys_path` temporary here, you can just fmt directly:

`fmt::format("{}{}{}", File::GetSysDirectory(), DIR_SEP, APPROVED_LIST_FILENAME)`